x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -157,35 +157,36 @@ else if (p.resolved()) {
 
     private static void checkAggregate(LogicalPlan p, Set<Failure> failures, AttributeMap<Expression> aliases) {
         if (p instanceof Aggregate agg) {
-
-            List<Expression> nakedGroups = new ArrayList<>(agg.groupings().size());
+            List<Expression> groupings = agg.groupings();
+            List<Expression> groups = new ArrayList<>(groupings.size());
             AttributeSet groupRefs = new AttributeSet();
             // check grouping
             // The grouping can not be an aggregate function
-            agg.groupings().forEach(e -> {
+            groupings.forEach(e -> {
                 e.forEachUp(g -> {
                     if (g instanceof AggregateFunction af) {
                         failures.add(fail(g, "cannot use an aggregate [{}] for grouping", af));
                     }
                 });
-                // remember naked expressions (to deal with implicit aliasing for grouping)
-                nakedGroups.add(e);
+                groups.add(e);
```

Comment:
I still think that `groups` and `groupings` are identical because we add each element of `groupings` to `group` here.

I asserted `groups.equals(groupings)` here and it at least doesn't fail any verifier or csv test.